From a0942b0409bcbc034ec26051a57c4137b887e823 Mon Sep 17 00:00:00 2001
From: Yi Kong <yikong@google.com>
Date: Wed, 1 Sep 2021 00:57:23 +0800
Subject: [PATCH] Do not use pragma diagnostics as a workaround

The current version of Harfbuzz contains unused code that triggers
compiler error. Since it's enabled through pragma, we cannot turn it off
through CFLAGS.

Turn off pragma diagnostics until we update to a newer version with the
code fix.

Test: build
Bug: 197333705
Change-Id: I6bcbe1fd751f6b9e753ee5e31c1bbdc759e97c62
---
 Android.bp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Android.bp b/Android.bp
index fa88ca6f..d69ed2bc 100644
--- a/Android.bp
+++ b/Android.bp
@@ -181,6 +181,7 @@ cc_library {
 
     export_include_dirs: ["src"],
     cflags: [
+        "-DHB_NO_PRAGMA_GCC_DIAGNOSTIC",
         "-DHAVE_INTEL_ATOMIC_PRIMITIVES",
         "-DHAVE_OT",
         "-DHAVE_ICU",
-- 
2.36.1

